egg-herbie/main.rkt

```diff
@@ -22,7 +22,33 @@
                  [else "libegg_math"])
                 (bytes->string/utf-8 (system-type 'so-suffix)))))
 
-(define-ffi-definer define-eggmath (ffi-lib libeggmath-path))
+; Checks if Racket is being run in emulation via rosetta
+(define (check-for-rosetta)
+  (equal? (with-output-to-string (lambda () (system "sysctl -n sysctl.proc_translated"))) "1\n"))

```

Comment:
I would move the tests for architecture and OS into here. For two reasons:

- Make this function safer—that is, it won't execute an unknown command if it is called on Linux or Windows.
- Logically, if you're not on macOS or you are running in ARM, you're definitely not in Rosetta